File: src/main/golang/leetcode/editor/cn/[230]Kth Smallest Element in a BST.go
package lc

//leetcode submit region begin(Prohibit modification and deletion)
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func kthSmallest(root *TreeNode, k int) (res int) {
	var j int
	var search func(node *TreeNode)
	search = func(node *TreeNode) {
		if node == nil {
			return
		}
		search(node.Left)
		j++
		if j == k {
			res = node.Val
			return
		}
		search(node.Right)
	}
	search(root)
	return
}

//leetcode submit region end(Prohibit modification and deletion)

File: src/main/golang/leetcode/editor/cn/[230]Kth Smallest Element in a BST.py
from typing import Optional
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
# leetcode submit region begin(Prohibit modification and deletion)
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        if root is None:
            return 0
        stack = []
        node = root
        while node is not None:
            stack.append(node)
            node = node.left
        while len(stack) > 0:
            node = stack.pop()
            k -= 1
            if k == 0:
                return node.val
            node = node.right
            while node is not None:
                stack.append(node)
                node = node.left
        return 0
# leetcode submit region end(Prohibit modification and deletion)
